Shooter tuning

hazeltorek opened the pull request.

JalenEvans committed:
Shooter changes

* Change up to speed to return 0 if the setpoint is 0 and the motor is
  currently also at 0.

* Moved conveyer sets into periodic instead of outtake methods.

* Used a conversion factor to convert from RPM to encoder units per 100
  ms.

Co-authored-by: Ell <ctorek@protonmail.ch>

hazeltorek committed:
Shooter tuning

* Move RPM error tolerance into Constants from Outtake subsystem and
  increase to 30 RPM.

* Change PID constants after tuning.

* Change outtake subsystem to decrease CAN bus traffic.

Co-authored-by: Jalen <jalenemmanuelevans@gmail.com>

hazeltorek committed:
Change RPM tolerance to 45

Was previously 30. This is the threshold the shooter velocity must be
within in order to enable the conveyor.

Co-authored-by: Jalen <jalenemmanuelevans@gmail.com>

JalenEvans reviewed (approved):
Very nice! I like the shooter code, it may just need a little more tuning in the future!
![Thumbs Up](https://user-images.githubusercontent.com/43455306/155814970-89535372-f73a-4677-9130-0fa4bf4cf70c.png)

jboutin020106 reviewed (approved):
Good job 👍

JalenEvans commented on src/main/java/frc/robot/subsystems/OuttakeSubsystem.java:
```diff
@@ -105,17 +112,54 @@ public double getSetpoint() {
+    return (setpoint * 2048) / 600;
+  }
+
+  private double getEncoderDelta() {
+    return encoder2.getIntegratedSensorPosition() + encoder1.getIntegratedSensorPosition();
+  }
+
+  public void resetEncoders() {
+    encoder1.setIntegratedSensorPosition(0, 10);
+    encoder2.setIntegratedSensorPosition(0, 10);
+  }
```
![No Java Docs](https://user-images.githubusercontent.com/43455306/155815162-a67e8ac4-24bf-4947-8581-6d88e5ab48d2.jpg)

VZamora5 reviewed (approved):
very cool

hazeltorek committed:
Added more comments to outtake

hazeltorek committed:
Added current limiting to the shooter motors

* Set the current limit threshold for the shooter motors to 50 A and the
  current limit after the threshold is triggered to 30 A, the breaker
  limit.

* Set the timeout for the current limit threshold to 0.5 seconds.

* Added all the current limit configuration values to the outtake
  constants.

JalenEvans reviewed (commented):
Thank you for the java docs!
![Thumbs Up](https://user-images.githubusercontent.com/43455306/155816806-905c2dc8-9d08-4566-bb9e-c759ed5a67c8.png)
